In [12]:
'''
Graded Challenge 5

Nama  : Nabila Sulistiowati
Batch : CODA-RMT-003

Program ini dibuat untuk menganalisis business knowledge dan menerapkan practical statistics pada sebuah dataset

'''

'\nGraded Challenge 5\n\nNama  : Nabila Sulistiowati\nBatch : CODA-RMT-003\n\nProgram ini dibuat untuk menganalisis business knowledge dan menerapkan practical statistics pada sebuah dataset\n\n'

In [13]:
# Import Libraries
import pandas as pd
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
from scipy.stats import ttest_ind
from scipy.stats import pearsonr, spearmanr

Business Understanding

A. Konteks Bisnis

Seblak adalah makanan khas Indonesia yang sedang viral dan menjadi tren kuliner. 
Sebagai seorang lulusan bootcamp Data Analytics, Saya akan mengeksplorasi potensi penjualan seblak di Tokopedia. 
Tujuannya adalah untuk memahami bagaimana masyarakat merespons produk ini dan menentukan apakah berjualan seblak bisa menjadi sumber pendapatan yang menguntungkan.

B. SMART Framework
1. Specific: Memahami performa penjualan dan preferensi konsumen terhadap produk seblak di Tokopedia.
2. Measurable: Menggunakan metrik seperti rata-rata penjualan, rating produk, serta pendapatan minimum dan maksimum.
3. Achievable: Analisis ini dapat dilakukan dengan dataset yang tersedia dan teknik statistik dasar.
4. Relevant: Hasil analisis akan membantu menentukan strategi bisnis yang optimal dalam menjual produk seblak.
5. Time-bound: Analisis dilakukan untuk periode data yang tersedia dan akan digunakan untuk keputusan dalam waktu 1 bulan ke depan.


In [14]:
# Import data
df = pd.read_csv('/Users/bee/Documents/Hacktiv8/GC/cleanedtokopedia_gc3.csv')
df.head()

Nama Produk  Harga  \
0                   Seblak Instan Premium Bangsatria  21000   
1  Paket Isi 4 Batagor Instan | Makaroni Basah | ...  64906   
2  GFS Seblak Basah Instan Original - Tinggal Sed...  16618   
3                   Kylafood Seblak Rempah Authentik  12900   
4  Kerupuk seblak kering kencur 500gram /seblak k...  28700   

                Penjual             Kota  Terjual  Rating  
0            Foodstocks  Jakarta Selatan       80     5.0  
1  Lakoca Official Shop           Cimahi       30     4.9  
2  Lakoca Official Shop           Cimahi      100     4.9  
3              kylafood          Bandung     3000     4.9  
4          oyun kriukzz    Jakarta Barat     1000     4.9

In [15]:
# Informasi data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 164 entries, 0 to 163
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Nama Produk  164 non-null    object 
 1   Harga        164 non-null    int64  
 2   Penjual      164 non-null    object 
 3   Kota         164 non-null    object 
 4   Terjual      164 non-null    int64  
 5   Rating       164 non-null    float64
dtypes: float64(1), int64(2), object(3)
memory usage: 7.8+ KB


In [16]:
# Menghitung rata-rata, median, standar deviasi, skewness, dan kurtosis masiing-masing kolom
statistics_desc = {
    'Harga': {
        'mean': df['Harga'].mean(),
        'median': df['Harga'].median(),
        'std_dev': df['Harga'].std(),
        'skewness': df['Harga'].skew(),
        'kurtosis': df['Harga'].kurt()
    },
    
    'Terjual': {
        'mean': df['Terjual'].mean(),
        'median': df['Terjual'].median(),
        'std_dev': df['Terjual'].std(),
        'skewness': df['Terjual'].skew(),
        'kurtosis': df['Terjual'].kurt()
    },
    
    'Rating': {
        'mean': df['Rating'].mean(),
        'median': df['Rating'].median(),
        'std_dev': df['Rating'].std(),
        'skewness': df['Rating'].skew(),
        'kurtosis': df['Rating'].kurt()
    }
}

# Menampilkan hasil statistik
print(statistics_desc)

{'Harga': {'mean': np.float64(25836.920731707316), 'median': np.float64(17500.0), 'std_dev': np.float64(35756.887951488075), 'skewness': np.float64(5.5229560608320245), 'kurtosis': np.float64(42.60994062057268)}, 'Terjual': {'mean': np.float64(410.3963414634146), 'median': np.float64(100.0), 'std_dev': np.float64(1143.929786869433), 'skewness': np.float64(6.071271140972659), 'kurtosis': np.float64(42.802453512404554)}, 'Rating': {'mean': np.float64(4.868292682926828), 'median': np.float64(4.9), 'std_dev': np.float64(0.21126440781040634), 'skewness': np.float64(-4.190896805523781), 'kurtosis': np.float64(24.162253144488638)}}


Dapat disimpulkan bahwa :
1.	Harga:
    - Rata-rata (Mean): Rp25,836.92
    - Median: Rp17,500
    - Standar Deviasi (Std Dev): Rp35,756.89
	- Skewness: 5.52 (Distribusi sangat skewed ke kanan, artinya ada beberapa produk dengan harga sangat tinggi)
	- Kurtosis: 42.61 (Distribusi sangat tajam, menunjukkan adanya banyak nilai yang jauh lebih tinggi dari rata-rata, mungkin karena beberapa produk dengan harga yang sangat mahal)

2. Terjual:
	- Rata-rata (Mean): 410.4
	- Median: 100
	- Standar Deviasi (Std Dev): 1,143.93
	- Skewness: 6.07 (Distribusi sangat skewed ke kanan, artinya ada beberapa produk yang terjual dalam jumlah yang sangat besar)
	- Kurtosis: 42.8 (Distribusi sangat tajam, menunjukkan adanya beberapa produk dengan penjualan yang sangat tinggi dibandingkan mayoritas lainnya)

3. Rating:
    - Rata-rata (Mean): 4.87
    - Median: 4.9
    - Standar Deviasi (Std Dev): 0.21
    - Skewness: -4.19 (Distribusi sangat skewed ke kiri, artinya sebagian besar produk mendapatkan rating yang sangat tinggi)
    - Kurtosis: 24.16 (Distribusi sangat tajam, banyak produk yang memiliki rating mendekati 5)


In [ ]:
# Menghitung pendapatan untuk setiap produk
df['revenue'] = df['Harga'] * df['Terjual']

# Rata-rata dan standar deviasi dari pendapatan
mean_revenue = df['revenue'].mean()
std_revenue = df['revenue'].std()

# Jumlah data (produk)
n = len(df)

# Z-score untuk confidence level 95%
z_score = 1.96

# Menghitung margin of error
margin_of_error = z_score * (std_revenue / np.sqrt(n))

# Menghitung confidence interval
lower_bound = mean_revenue - margin_of_error
upper_bound = mean_revenue + margin_of_error

lower_bound, upper_bound

(np.float64(3777519.185231424), np.float64(15337379.205012478))

Hasil perhitungan confidence interval untuk pendapatan bulanan dari penjualan produk seblak adalah sebagai berikut:

- Lower Bound (Batas Bawah): Rp3,777,519.19
- Upper Bound (Batas Atas): Rp15,337,379.21

Kesimpulan:
- Lower Bound menunjukkan estimasi pendapatan terendah dengan tingkat kepercayaan 95%.
- Upper Bound menunjukkan estimasi pendapatan tertinggi dengan tingkat kepercayaan 95%.


In [18]:
# Memisahkan data berdasarkan lokasi
jabodetabek = df[df['Kota'].str.contains("Jakarta|Bekasi|Depok|Bogor|Tangerang")]['Harga']
luar_jabodetabek = df[~df['Kota'].str.contains("Jakarta|Bekasi|Depok|Bogor|Tangerang")]['Harga']

# Melakukan uji t dua sampel independen
t_stat, p_value = ttest_ind(jabodetabek, luar_jabodetabek)

t_stat, p_value

(np.float64(0.49477799420796204), np.float64(0.6214271917158096))

Hasil uji t menunjukkan nilai statistik t sekitar 0.495 dan p-value sekitar 0.621. Yang dapat disimpulkan : 
- Nilai p-value (0.621):
- Karena p-value lebih besar dari 0.05, hipotesis nol (H0) diterima.
- Ini menunjukkan bahwa tidak ada bukti yang cukup untuk menyatakan bahwa terdapat perbedaan signifikan dalam harga barang antara Jabodetabek dan luar Jabodetabek pada tingkat signifikansi 5%.

Kesimpulan:

Tidak ada perbedaan yang signifikan dalam harga barang di Jabodetabek dan luar Jabodetabek berdasarkan data yang tersedia. Hal ini mengindikasikan bahwa biaya bahan baku atau faktor lainnya mungkin tidak berpengaruh besar terhadap perbedaan harga di kedua daerah tersebut.

In [19]:

# Analisis korelasi harga dan rating
harga = df['Harga'].dropna()    # Data harga produk
rating = df['Rating'].dropna()  # Data rating produk

# Menggunakan korelasi Pearson
correlation, p_value = pearsonr(harga, rating)

# Output hasil korelasi dan p-value
print("Korelasi: ", correlation)
print("P-value: ", p_value)

# Interpretasi hasil
if p_value < 0.05:
    print("Korelasi signifikan secara statistik.")
else:
    print("Tidak ada korelasi signifikan.")

Korelasi:  0.006017721941190796
P-value:  0.9390406608512415
Tidak ada korelasi signifikan.


Dapat disimpulkan bahwa:
- Korelasi: 0.006: Hubungan linear antara harga produk dan kesukaan sangat lemah, hampir tidak ada.
- P-value: 0.939: Jauh di atas 0.05, sehingga hasil ini tidak signifikan secara statistik.

Tidak ada korelasi signifikan antara harga produk dan kesukaan terhadap produk. Ini berarti harga produk dalam dataset ini tidak memengaruhi seberapa banyak orang menyukai produk tersebut secara statistik dan terdapat kemungkinan terdapat faktor lain yang lebih berpengaruh dalam kecenderungan orang membeli seblak.